### PR TITLE
Add checkstyle support to the realm sub-project

### PR DESCRIPTION
**Pull request by emanuelez**

### CODE REVIEW

File: realm/config/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
+    "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">
+
+<module name="Checker">
+    <!-- Files to ignore -->
+    <module name="SuppressionFilter">
+        <property name="file" value="${proj.module.dir}/config/checkstyle/checkstyle-suppressions.xml"/>
+    </module>
+
+    <!-- TODO: UNCOMMENT
```
Comment (bmunkholm) [inline realm/config/checkstyle/checkstyle.xml]: Can't we enable this? There are a few misses - including in this PR! :-)

File: realm/config/checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    <!-- Trailing spaces -->
+    <!-- TODO: UNCOMMENT
+    <module name="RegexpSingleline">
+        <property name="format" value="\s+$"/>
+        <property name="message" value="Line has trailing spaces."/>
+    </module> -->
+
+    <!-- Space after 'for' and 'if' -->
+    <module name="RegexpSingleline">
+        <property name="format" value="^\s*(for|if)[^ ]"/>
```
Comment (bmunkholm) [inline realm/config/checkstyle/checkstyle.xml]: add "while", "do" - others?

File: realm/config/checkstyle/checkstyle.xml [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    <!-- Trailing spaces -->
+    <!-- TODO: UNCOMMENT
+    <module name="RegexpSingleline">
+        <property name="format" value="\s+$"/>
+        <property name="message" value="Line has trailing spaces."/>
+    </module> -->
+
+    <!-- Space after 'for', 'if', 'while' -->
+    <module name="RegexpSingleline">
+        <property name="format" value="^\s*(for|if|while)[^ ]"/>
```
Comment (cmelchior) [inline realm/config/checkstyle/checkstyle.xml]: Also ```do {```

**Comment by cmelchior** [comment]
How easy/hard would it be to make a IntelliJ style file for the same rules?

File: realm/config/checkstyle/checkstyle.xml (hunk 3 above)
Comment (emanuelez) [inline realm/config/checkstyle/checkstyle.xml]: Do is different and is handled at line 94

**Comment by emanuelez** [comment]
Bump @realm/java

File: realm/src/main/java/io/realm/internal/RealmCore.java [hunk 4]
```diff
@@ -101,7 +101,7 @@ public static void gcGuaranteed(){
         Object obj = new Object();
         WeakReference<Object> ref = new WeakReference<Object>(obj);
         obj = null;
-        while(ref.get()!=null)
+        while (ref.get()!=null)
```
Comment (cmelchior) [inline realm/src/main/java/io/realm/internal/RealmCore.java]: We should also add a rule so there is whitespace inside the () 
```while (ref.get() != null)```

**Comment by cmelchior** [comment]
Apart from comment :+1:

File: realm/src/main/java/io/realm/internal/RealmCore.java (hunk 4 above)
Comment (beeender) [inline realm/src/main/java/io/realm/internal/RealmCore.java]: Need a space inside the ()? Really? Or prevent from there is a space inside ()?

Comment (cmelchior) [inline realm/src/main/java/io/realm/internal/RealmCore.java]: You don't like spaces around the operators?
```while (ref.get() != null)```
vs
```while (ref.get()!=null)```

The first is a lot more readable imo

Comment (beeender) [inline realm/src/main/java/io/realm/internal/RealmCore.java]: Ha, never mind. I thought you were saying need a space in `ref.get()` like `ref.get( )`

**Comment by emanuelez** [comment]
@cmelchior Are you ready to review the output of a global `Alt ⌘ L` ? :)

**Comment by cmelchior** [comment]
Hit me ☺

**Comment by emanuelez** [comment]
Hehe, no :) I want to keep this PR small and reviewable. Once the system is in place I will make another big PR with all the automatic fixes.

**Comment by emanuelez** [comment]
Rebump @realm/java

**Comment by beeender** [comment]
:+1:

**Comment by kneth** [comment]
:+1:

**Pull request merged**